CONTRIBUTING.md

```diff
@@ -6,13 +6,27 @@ feedback and contributions from the community.
 Reviewing this document will maximize your success in working with the
 codebase and sending pull requests.
 
+## Quick Links
+
+-   [How to Run the extension](#run)
+-   [Development Guidelines](#guidelines)
+-   [More detailed docs](./docs)
+-   [Telemetry](./docs/telemetry.md)
+-   [Using the `Commands` class](./docs/ARCHITECTURE.md#commands)
+
 ## Getting Started
 
-### Find things to do
+### Extension development flow
 
-If you're looking for ideas about where to contribute, consider
-[_good first issue_](https://github.com/aws/aws-toolkit-vscode/labels/good%20first%20issue)
-issues.
+This explains the high level flow of going from nothing to a new release with your change. The actual details are explained in the rest of the document.
```

Comment:
is this section answering questions we've had before? it seems very general, i'm wondering if it would help me as a new contributor.